compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -467,6 +467,17 @@ private void checkXMLNamespacePrefixes(List<BLangXMLElementFilter> filters) {
         }
     }
 
+    private int getPreferredMemberTypeTag(BFiniteType finiteType) {
+        for (BLangExpression valueExpr : finiteType.getValueSpace()) {
+            for (int i = TypeTags.INT; i <= TypeTags.DECIMAL; i++) {
```

Comment:
let's continue to next `valueExpr` if `valueExpr.getBType().tag` not within Int/Decimal range.
If not all cases would have to loop through inner loop as well